demo/src/Components/FormElements/TextInput.jsx

```diff
@@ -16,6 +16,13 @@ const TextInput = () => (
         <label htmlFor="text-input-3" className="c-label">Message:</label>
         <textarea id="text-input-3" className="c-text-input c-text-input--textarea" placeholder="Your message…" />
       </div>
+      <div className="o-grid__item u-1/2@sm">
+        <label htmlFor="text-input-4" className="c-label">Search:</label>
+        <div className="c-text-search">
+          <input type="search" id="text-input-4" name="search" placeholder="Search..." className="c-text-input c-text-input--search" />
+          <button type="submit" className="c-btn c-btn--alpha c-btn--md c-btn-icon c-btn-icon--search" />
```

Comment:
We can maybe look into combining a few of these... but I can see what you are trying to do.  I think the Icon could be come a component in it's own right that sits inside of the button.   The button just being a container for text and an optional icon perhaps.